akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala

```diff
@@ -508,7 +512,7 @@ final class Partition[T](outputPorts: Int, partitioner: T ⇒ Int) extends Graph
         val elem = grab(in)
         val idx = partitioner(elem)
         if (idx < 0 || idx >= outputPorts)
-          failStage(new IndexOutOfBoundsException(s"partitioner must return an index in the range [0,${outputPorts - 1}]. returned: [$idx] for input [$elem]."))
+          failStage(PartitionOutOfBoundsException(s"partitioner must return an index in the range [0,${outputPorts - 1}]. returned: [$idx] for input [$elem]."))
```

Comment:
hmm, this should probably not include the element—we might want to perform an audit that ensure that we do not log `elem.tostring` anywhere (because that could actually be HUGE)